string[] args
---
string msg
bool success
